### PR TITLE
Update Envoy to 1ae6fb2 (Apr 8th, 2024)

### PR DESCRIPTION
**Pull request by yanavlasov**
- Adjust code to changes in the createTagProducer return value.

### CODE REVIEW

File: source/client/process_impl.cc [hunk 1]
```diff
@@ -848,8 +848,14 @@ bool ProcessImpl::runInternal(OutputCollector& collector, const UriPtr& tracing_
     // Needs to happen as early as possible (before createWorkers()) in the instantiation to preempt
     // the objects that require stats.
     if (!options_.statsSinks().empty()) {
-      store_root_.setTagProducer(
-          Envoy::Config::StatsUtility::createTagProducer(bootstrap_, envoy_options_.statsTags()));
+      auto producer_or_error = Envoy::Stats::TagProducerImpl::createTagProducer(
```
Comment (mum4k) [inline source/client/process_impl.cc:851]: Would you mind listing the full type here instead of the `auto`? The type isn't obvious to me from the context.

**Review by mum4k — APPROVED** [review]
Thank you!

File: source/client/process_impl.cc (hunk 1 above)
Comment (yanavlasov) [reply source/client/process_impl.cc]: Done

Comment (mum4k) [reply source/client/process_impl.cc]: Thank you.

**Review by mum4k — APPROVED** [review]

**Pull request merged**